src/Parser/Warrior/Arms/Modules/Talents/Rend.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import Wrapper from 'common/Wrapper';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import Analyzer from 'Parser/Core/Analyzer';
+
+const EXECUTE_RANGE = 0.2;
+
+class RendAnalyzer extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+  };
+
+  rendsInExecuteRange = 0;
+  targets = [];
+  failedTargets = [];
+
+  on_initialized() {
+		this.active = this.combatants.selected.hasTalent(SPELLS.REND_TALENT.id);
+	}
+
+  on_byPlayer_cast(event) {
+    if(SPELLS.REND_TALENT.id === event.ability.guid) {
+      // If Rend is used flag the target to check their health on the next damage event.
+      this.targets.push(event.targetID);
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const targetId = event.targetID;
+
+    if(SPELLS.REND_TALENT.id === event.ability.guid && this.targets.indexOf(targetId) !== -1) {
+      // Check if the target's health was within execute range.
+      if(event.hitPoints / event.maxHitPoints < EXECUTE_RANGE) {
+        if(this.failedTargets.indexOf(targetId) === -1) {
+          // If this is the first time Rend was used on this target in execute range we flag that target.
+          // This is to prevent false positives from Rend being applied just as the target enters execute range.
```

Comment:
Are you sure these casts shouldn't just be counted as bad aswell? I have no idea what the threshold is as I don't play warrior, but since this check always happens on the first tick of the Rend I would imagine it might be just as bad as casting it druing execute range since it's rage you could've spent on execute anyway.

Reply:
alright sounds reasonable.